src/Indicator.vala

```diff
@@ -197,6 +202,15 @@ public class Power.Indicator : Wingpanel.Indicator {
         string? primary_text = null;
         string? secondary_text = null;
         if (display_device != null) {
+            if (display_device.percentage == LOW_BATTERY_PERCENTAGE && !display_device.is_charging) {
+                is_showing_low_battery_percent = true;
+                settings.set_boolean ("show-percentage", true);
+            }
+            /* Hide low battery percentage after plug charger */
+            if (is_showing_low_battery_percent && display_device.is_charging) {
+                settings.set_boolean ("show-percentage", false);
```

Comment:
If the user has deliberately set the "show-percentage" setting to "true", this will override the user's intention of always showing the intention.  I think you need to find a way of showing the percentage when the battery is low  *despite* the user setting , rather than changing the user setting.

Its probably not a good idea to remember the original setting because the user may change it while the batter is low.